### PR TITLE
Release v2.0.0

### PR DESCRIPTION
**Pull request by Fujikawas**
## Description
The dumux-adapter comes to v2.0.0 to work with preCICE v3.0.0 and DuMuX from version 3.7. Outdated information are removed.

## Checklist

- [ ] I made sure that the source files are formatted properly.
- [ ] I added my changes to the changelog (`CHANGELOG.md`)
- [ ] I updated the documentation.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -21,13 +21,13 @@ The versions of the main dependencies can be manipulated when building the Docke
...
 - `ARG DUMUXVERSION=3.7`: DuMuX version. Available in all recipes.
-- `ARG PRECICEVERSION=develop`: preCICE version. Available only in "large" and "slim" recipes.
+- `ARG PRECICEVERSION=3.0.0`: preCICE version. Available only in "large" and "slim" (use branch name e.g. "v3.0.0" or "develop") recipes.
 - `ARG PRECICEUBUNTU=focal`: preCICE target platform (here, Ubuntu Focal Fossa). Available only in "large" recipe.
 
 If one wants to build a "slim" image using DUNE 2.9, DuMuX from the current master branch and preCICE 3.0.0 one could call it with the following command:
 
 ```text
-sudo docker build --build-arg DUNEVERSION=2.9 --build-arg DUMUXVERSION=master --build-arg PRECICEVERSION=3.0.0 -t precice/dumux-precice:master-3.0.0 --file dockerfile.slim .
+sudo docker build --build-arg DUNEVERSION=2.9 --build-arg DUMUXVERSION=master --build-arg PRECICEVERSION=v3.0.0 -t precice/dumux-precice:master-v3.0.0 --file dockerfile.slim .
```
Comment (mathiskelm) [inline docker/README.md:30]: @IshaanDesai the workflow file would need versions such as "v3.0.0" which would change the tag pattern but the alternative is adding a "develop or release version" check as in `checkout_repositories.sh`. Any thoughts?

Comment (IshaanDesai) [reply docker/README.md:30]: Which workflow file are you referring to here?

Comment (mathiskelm) [reply docker/README.md:30]: The two .slim dockerfiles use `PRECICEVERSION` as a branch/tag name (to allow for "develop"). In the .large dockerfile the variable is used to select a .deb package which doesn't include a "v".

Comment (IshaanDesai) [reply docker/README.md:30]: Ah okay. This is indeed inconsistent. Do we even use the `.large` files anywhere at all?

Comment (mathiskelm) [reply docker/README.md:30]: I can only see the two .slim recipes being used.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # DuMuX-preCICE change log
 
-## Not released yet
+## v2.0.0
+
+- 2023-08-10: Updated the adapter to align with preCICE V3. Updated the examples accordingly.
```
Comment (IshaanDesai) [inline CHANGELOG.md:5]:
```suggestion
- 2023-08-10: Updated the adapter to be compatible with preCICE v3.0.0. Updated the examples too.
```

File: examples/dummysolver/precice-dummy-solver-config.xml [hunk 3]
```diff
@@ -1,73 +1,66 @@
 <?xml version="1.0"?>
 
-<precice-configuration>
+<precice-configuration >
```
Comment (IshaanDesai) [inline examples/dummysolver/precice-dummy-solver-config.xml:3]: ?

**Review by IshaanDesai — APPROVED** [review]

File: docker/README.md (hunk 1 above)
Comment (IshaanDesai) [reply docker/README.md:30]: > I can only see the two .slim recipes being used.

In that case, I would suggest opening an issue about this inconsistency and finalizing the release for now.

Comment (mathiskelm) [reply docker/README.md:30]: See #33.

**Review by mathiskelm — APPROVED** [review]

**Pull request merged**